src/app/services/fake-goal-management-service.ts

```diff
@@ -0,0 +1,54 @@
+import {Goal} from '../model/goal';
+
+export class FakeGoalManagementService {
+  goal: Goal;
+  goals = new Map<string, Goal>();
+  userGoals = new Map<string, Goal[]>();
+
+  constructor() {}
+
+  setGoal(
+      name: string, startDate: Date, endDate: Date, workload: number,
+      avgWorkload: number) {
+    let id = this.goals.size;
+    let diffDays =
+        (endDate.valueOf() - startDate.valueOf()) / (1000 * 3600 * 24);
+    let dailyProgress = new Array<number>(diffDays);
```

Comment:
Avoid using the Array constructor. It has confusing usage:
const x = Array(2) // [undefined, undefined]
const y = Array(2, 3) // [2, 3]

In your case, you can use:
Array.from<number>({length: diffDays}).fill(0);

Reply:
ok. I have changed it and keep the groups variable same as before because the groups list doesn't have the default value nor the fixed length.

Reply:
another problem is that when I changed the initialization of dailyProgress, there would be an error saying Type unknown[] is not assignable to type 'number[]'.